packages/flutter/lib/src/material/toggleable.dart

```diff
@@ -268,29 +268,19 @@ abstract class RenderToggleable extends RenderConstrainedBox implements Semantic
   @override
   Iterable<SemanticAnnotator> getSemanticAnnotators() sync* {
     yield (SemanticsNode semantics) {
-      semantics.hasCheckedState = true;
-      semantics.isChecked = _value;
-      semantics.canBeTapped = isInteractive;
+      semantics
+        ..hasCheckedState = true
+        ..isChecked = _value;
+      if (isInteractive)
+        semantics.addAction(SemanticAction.tap);
```

Comment:
add tests for what happens if something switches from being interactive to not being interactive.
You're changing the semantics here in a very meaningful way: this class now requires that its node be reset if it changes from interactive to not interactive, so that the old action is no longer present.